### PR TITLE
Implement food

### PR DESCRIPTION
**Pull request by bumjung**

### CODE REVIEW

File: modules/food.js [hunk 1]
```diff
@@ -1,5 +1,53 @@
-function Food()
+
+function Food(x, y, energy)
 {
+	this.location = new Vector(x, y);
```
Comment (insoochoo) [inline modules/food.js]: nit: indentation

File: modules/food.js [hunk 2]
```diff
@@ -1,5 +1,53 @@
...
+  // randomize direction
+  this.velocity.rotate(Math.PI * 2 * Math.random());
 }
 
-Food.prototype = {}
+Food.prototype = {
+	draw: function(ctx)
+	{
+		ctx.beginPath();
+		ctx.globalAlpha = 0.5;
```
Comment (insoochoo) [inline modules/food.js]: nit: indentation

File: modules/food.js [hunk 3]
```diff
@@ -1,5 +1,53 @@
-function Food()
+
+function Food(x, y, energy)
 {
+	this.location = new Vector(x, y);
+  this.velocity = new Vector(.5, .5);
+  this.energy = energy;
+  this.dead = false;
```
Comment (insoochoo) [inline modules/food.js]: since this is food attribute, maybe we should call it depleted?

Comment (bumjung) [inline modules/food.js]: good point, done

File: modules/food.js [hunk 4]
```diff
@@ -1,5 +1,53 @@
...
+
+  this.radius = this.energy * this.RADIUS_RATIO;
+
+  // randomize direction
+  this.velocity.rotate(Math.PI * 2 * Math.random());
 }
 
-Food.prototype = {}
+Food.prototype = {
+	draw: function(ctx)
```
Comment (insoochoo) [inline modules/food.js]: this still needs indentation fix to 2 spaces

File: modules/food.js [hunk 5]
```diff
@@ -1,5 +1,53 @@
...
+    ctx.stroke();
+
+    ctx.font = "15px sans-serif";
+    ctx.fillStyle = "black";
+    ctx.fillText("FOOD", this.location.x - 23, this.location.y + 5);
+	},
+
+	update: function(land)
+	{
+		this.depleted = this.energy <= 0;
```
Comment (insoochoo) [inline modules/food.js]: same thing

File: modules/food.js [hunk 6]
```diff
@@ -1,5 +1,53 @@
...
+		var newRadius = this.energy > 0 ? this.energy * this.RADIUS_RATIO : 0;
+
+		// decrease radius gradually
+		this.radius += (newRadius - this.radius) / 5;
+
+		this.location.add(this.velocity);
+
+		// kill the food if out of bounds
+    if (this.location.x < -10 || this.location.x > land.width + 10 ||
+			this.location.y < -10 || this.location.y > land.height + 10)
```
Comment (insoochoo) [inline modules/food.js]: if (xxxx ||
---yyyy)

File: modules/food.js [hunk 7]
```diff
@@ -1,5 +1,53 @@
...
+    {
+			this.energy = 0;
+    }
+	}
```
Comment (insoochoo) [inline modules/food.js]: here

File: modules/simulator.js [hunk 8]
```diff
@@ -33,6 +38,24 @@ function step(ctx, land)
...
+    var food = land.food[i];
+
+    if (!food.depleted)
+    {
+      food.update(land);
+      food.draw(ctx);
+    }
+    else
+    {
+      if(Math.random() <= .005)
```
Comment (insoochoo) [inline modules/simulator.js]: should we move all probability related values into constants?

**Comment by insoochoo** [comment]
:+1:
:100:

**Pull request merged**